pallets/avn-proxy/src/lib.rs

```diff
@@ -126,8 +134,19 @@ pub mod pallet {
             let call_hash: T::Hash = T::Hashing::hash_of(&call);
             let sender: T::Origin = frame_system::RawOrigin::Signed(proof.signer.clone()).into();
 
-            call.dispatch(sender).map(|_| ()).map_err(|e| e.error)?;
-            Self::deposit_event(Event::<T>::CallDispatched { relayer, hash: call_hash });
+            let dispatch_result = call.dispatch(sender).map(|_| ()).map_err(|e| e.error);
+            match dispatch_result {
+                Ok(_) => {
+                    Self::deposit_event(Event::<T>::CallDispatched { relayer, hash: call_hash });
```

Comment:
I would go with `DispatchFailed`. 

But more generally, should we go with 2 events or 1 event with a status field? I know we went with the 2 events approach for EthereumEvents but just asking in case 1 event is simpler to check.

Reply:
I would use `InnerCallDispatchFailed` or `InnerCallFailed`. I think the former is equivalent grammatically to what Nahu suggested, but could give more context. I am not sure we can Dispatch anything else other than an inner call, so I won't make a fuss between the two.

Reply:
I don't have a strong preference either, its just that `innerCall` is a bit redundant because there is nothing else we are dispatching in this pallet.

Reply:
I think DispatchFailed is an overloaded term, hence saying having the Inner part.
If you are both happy with InnerCallFailed I can go with that.
Regarding using one or two, if we go with one event, we are changing also an existing interface, which I was trying to avoid. Then the error field will have to become an optional field.
I think having a separate one will be cleaner.